lib/gears/surface.lua

```diff
@@ -45,17 +45,23 @@ function surface.load_uncached_silently(_surface, default)
     if cairo.Surface:is_type_of(_surface) then
         return _surface
     end
+
+    local err, s
+
     -- Strings are assumed to be file names and get loaded
     if type(_surface) == "string" then
-        local err
         file = _surface
-        _surface, err = capi.awesome.load_image(file)
+
+        -- `s` is an undocumented workaround until #908 is implemented
+        _surface, err, s = capi.awesome.load_image(file)
+
         if not _surface then
             return get_default(default), err
         end
     end
+
     -- Everything else gets forced into a surface
-    return cairo.Surface(_surface, true)
+    return s or cairo.Surface(_surface, true)
```

Comment:
Can't the shims just override `gears.surface` instead? I.e. `require("gears.surface").foo = my_shim`?
That way, no hacks end up "in the real code".